File: AddressBook/AddressBook/Dtos/City.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class CityRes
    {
        public int CityId { get; set; }
        public string? Name { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        // Foreign keys
        public int? StateId { get; set; }
        public State? State { get; set; }
    }

    public class CityReqEdit
    {
        [Required]
        public string? Name { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public int? StateId { get; set; }
    }

    public class CityReqSearch : PagedReq
    {
        public int? StateId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/City.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";
import { StateRes } from "./State";

export interface CityRes {
  cityId?: string;
  name?: string;
  latitude?: number;
  longitude?: number;

  stateId?: string;
  state?: StateRes;
}

export class CityReqEdit {
  name?: string = "";
  latitude?: number = 0;
  longitude?: number = 0;

  stateId?: string = "";
  constructor(stateId?: string) {
    this.stateId = stateId;
  }
}

export class CityReqSearch extends PagedReq {
  stateId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {stateId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.stateId = stateId;
  }
}